docs/function-mesh-worker/manage-builtin-connectors.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Manage the built-in connectors
+category: function-mesh-worker
+id: manage-builtin-connectors
+---
+
+This document describes how to manage the built-in Connectors with Function Mesh Worker service. 
+
+## Built-in Connector feature in Apache Pulsar
+
+Pulsar distribution includes a set of common connectors that have been packaged and tested with the rest of Apache Pulsar. These connectors import and export data from some of the most commonly used data systems.
+
+Using any of these connectors is as easy as writing a simple connector and running the connector locally or submitting the connector to a Pulsar Functions cluster.
+
+Pulsar Functions Worker provides `sinks` and `sources` APIs to let the user manage Pulsar IO Connectors, it also provides a built-in connectors feature, with this feature, user do not need to provide the connectors' NAR package file and uses the connector type directly. From the Pulsar Functions Worker side, it will load the packages from `connectorsDirectory` (defaults to `./connectors`) and uses the connectors as built-in connectors list.
+
+## Customize the built-in connectors in Function Mesh Worker Service
+
+With Function Mesh Worker service, it requires a customized YAML file placed into `conf/connectors.yaml` with content of the metadata of each built-in connector. Below are the configureable fields of built-in connector:
+
+| Field | Description |
+| ---|---|
+| `id` | The id of the IO connector. |
+| `name` | The name of the connector type. |
+| `description` | Description to be used for user help. |
+| `sourceClass` | The class name for the connector source implementation. If not defined, it will be assumed this connector cannot act as a data source. |
+| `sinkClass` | The class name for the connector sink implementation. If not defined, it will be assumed this connector cannot act as a data sink. |
+| `sourceConfigClass` | The class name for the source config implementation. If not defined, the framework will not be able to do any submission time checks. |
+| `sinkConfigClass` | The class name for the sink config implementation. If not defined, the framework will not be able to do any submission time checks. |
+| `version` | The version of the connector. |
+| `imageRegistry` | The imageRegistry where host the connector image. By default the imageRegistry is empty, which refer to Docker Hub `docker.io/`. |
+| `imageRepository` | The imageRepository to the connector. Usually it in format of NAMESPACE/REPOSITORY. |
+| `imageTag` | The imageTag to the connector image. By default it will align with Pulsar's version. |
+| `typeClassName` | Type class name of the connector or function. If not set, the default value '[B' will be used |
+| `sourceTypeClassName` | Type class name of the source connector. If not set, will inherit the value from typeClassName. |
```

Comment:
```suggestion
| `sourceTypeClassName` | The type class name of the source connector. If not set, it will inherit the value from the `typeClassName` field. |
```